pkg/providers/vsphere/virtualmachine/cdrom.go

```diff
@@ -235,6 +214,63 @@ func getBackingFileNameByImageRef(
 	return itemStatus.FileInfo[0].StorageURI, nil
 }
 
+// processImage validates if the image is ready and of type ISO, and returns the
+// content library item UUID and status from its provider ref.
+func processImage(
+	ctx context.Context,
+	client ctrlclient.Client,
+	imageName string,
+	namespace string) (string, imgregv1a1.ContentLibraryItemStatus, error) {
+
+	var vmi vmopv1.VirtualMachineImage
+
+	if namespace != "" {
+		// Namespace scope image.
+		if err := client.Get(ctx, ctrlclient.ObjectKey{Name: imageName, Namespace: namespace}, &vmi); err != nil {
+			return "", imgregv1a1.ContentLibraryItemStatus{}, err
+		}
+	} else {
+		// Cluster scope image.
+		var cvmi vmopv1.ClusterVirtualMachineImage
+		if err := client.Get(ctx, ctrlclient.ObjectKey{Name: imageName}, &cvmi); err != nil {
+			return "", imgregv1a1.ContentLibraryItemStatus{}, err
+		}
+		vmi = vmopv1.VirtualMachineImage(cvmi)
+	}
+
+	// Verify image before retrieving content library item.
+	if !conditions.IsTrue(&vmi, vmopv1.ReadyConditionType) {
```

Comment:
I presume VMI controller already watches the item Ready condition, which won't be true if the item Ready condition is false.

Reply:
Yep, it's based on both the item's ready condition and security compliance: https://github.com/vmware-tanzu/vm-operator/blob/main/controllers/contentlibrary/contentlibraryitem/contentlibraryitem_controller.go#L192-L214